packages/core/src/middleware/koa-spa-session-guard.ts

```diff
@@ -55,9 +55,12 @@ export default function koaSpaSessionGuard<
           throw new RequestError({ code: 'session.not_found', status: 404 });
         }
 
-        ctx.redirect(
-          appendPath(getTenantEndpoint(tenantId, EnvSet.values), sessionNotFoundPath).href
-        );
+        const tenantEndpoint = getTenantEndpoint(tenantId, EnvSet.values);
+        // Replace to current hostname (if custom domain is used)
+        // eslint-disable-next-line @silverhand/fp/no-mutation
+        tenantEndpoint.hostname = ctx.request.hostname;
```

Comment:
Avoid direct mutation of the hostname?

Reply:
Unfortunately, there is no better way for `URL` instance?